include/solver.h

```diff
@@ -241,6 +247,61 @@ class AbsSmtSolver
   /* Reset all assertions */
   virtual void reset_assertions() = 0;
 
+
+  /* Initialize a datatype declaration with some name
+   * @param s Name of the datatype
+   * @return an empty Datatype declaration
+   */
+  virtual DatatypeDecl make_datatype_decl(const std::string & s) = 0;
+
+  /* Initialize a datatype constructor declaration with some name
+   * @param s Name of the datatype constructor
+   * @return an empty Datatype declaration
+   */
+  virtual DatatypeConstructorDecl make_datatype_constructor_decl(const std::string s) const = 0; // what is const=0?
+
+  /* Add a datatype constructor to a datatype declaration
+   * @param dt Datatype
+   * @param con Datatype constructor
+   */
+ virtual void addConstructor(DatatypeDecl & dt, const DatatypeConstructorDecl & con) const = 0; // what is const=0?
```

Comment:
I left comments about the naming scheme of methods in the CVC4 implementations, but of course that applies to the abstract interface and any other implementations. I won't duplicate those comments any more.